dask/array/core.py

```diff
@@ -1080,6 +1080,16 @@ def numblocks(self):
     def npartitions(self):
         return reduce(mul, self.numblocks, 1)
 
+    def persist(self):
+        return super().persist()
+
+    def compute_metadata(self):
```

Comment:
Given that this function is user facing it should probably have a full docstring describing the full behavior.  I would expect a single line explanation, followed up by probably a few sentences describing when this situation comes about, why it's necessary, and any other details, like in-place modification.